library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala

```diff
@@ -50,3 +41,18 @@ sealed trait EnvironmentInfo extends js.Object {
    */
   def javaSystemProperties: js.UndefOr[js.Dictionary[String]]
 }
+
+object EnvironmentInfo {
+  /** The value of the `__ScalaJSEnv` variable, if it exists. */
+  val envInfo: js.UndefOr[EnvironmentInfo] = {
+    import js.Dynamic.{global => g}
+
+    // We've got to use selectDynamic explicitly not to crash Scala 2.10
+    if (js.typeOf(g.selectDynamic("__ScalaJSEnv")) == "object" &&
+        js.DynamicImplicits.truthValue(g.selectDynamic("__ScalaJSEnv"))) {
```

Comment:
Indeed. It was a direct translation of the JS code and I haven't given it any thought. Comparing to `null` would be better indeed.